pkg/rancher-desktop/components/MarketplaceCard.vue

```diff
@@ -1,39 +1,139 @@
+        <i class="icon icon-external-link " />
+      </a>
+    </div>
+    <div class="extensions-card-footer">
+      <button
+        data-test="button-install"
+        :class="installed ? 'role-danger': 'role-primary'"
+        class="btn btn-xs"
+        :disabled="loading"
+        @click="appInstallation(installationAction)"
+      >
+        <span v-if="loading" name="loading" :is-loading="loading">
+          <loading-indicator>{{ buttonLabel }}</loading-indicator>
+        </span>
+        <span v-if="!loading">{{ buttonLabel }}</span>
+      </button>
     </div>
-    <div class="extension-card-footer"></div>
   </div>
 </template>
 
 <script>
+
+// TODO: icon to external link
+import LoadingIndicator from '@pkg/components/LoadingIndicator.vue';
+import demoMetadata from '@pkg/utils/_demo_metadata.js';
 export default {
-  props: {
+  components: { LoadingIndicator },
+  props:      {
     extension: {
       type:     Object,
       required: true,
     },
+    credentials: {
+      type:     Object,
+      required: true,
+    },
+    isInstalled: {
+      type:     Boolean,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      loading:          false,
+      installed:        null,
+      extensionDetails: null,
+      error:            null,
+      response:         null,
+      bannerActive:     false,
+    };
   },
+  computed: {
+    installationAction() {
+      return this.installed ? 'uninstall' : 'install';
+    },
+    versionedExtension() {
+      return `${ this.extensionWithoutVersion }:${ this.extensionDetails?.version }`;
+    },
+    extensionWithoutVersion() {
+      const index = this.extension.slug.lastIndexOf(':');
 
+      return this.extension.slug.substring(0, index) || this.extension.slug;
+    },
+    extensionLink() {
+      return this.extension.slug.includes('ghcr.io') ? `https://${ this.extension.slug }` : `https://hub.docker.com/extensions/${ this.extension.slug }`;
+    },
+    buttonLabel() {
+      if (this.loading) {
+        return this.installed ? this.t('marketplace.sidebar.uninstallButton.loading') : this.t('marketplace.sidebar.installButton.loading');
+      } else {
+        return this.installed ? this.t('marketplace.sidebar.uninstallButton.label') : this.t('marketplace.sidebar.installButton.label');
+      }
+    },
+  },
+
+  mounted() {
+    this.metadata = demoMetadata[this.extensionWithoutVersion];
+
+    if (!this.metadata) {
+      return;
+    }
+
+    this.installed = this.isInstalled;
+    this.extensionDetails = {
+      name:
+        this.metadata?.LatestVersion.Labels['org.opencontainers.image.title'] ||
+        this.extensionWithoutVersion,
+      version: this.metadata?.LatestVersion.Tag || [],
+    };
+  },
   methods: {
-    route() {
-      this.$router.push({
-        name:   'marketplace-details',
-        params: {
-          slug:  this.extension.slug,
-          image: this.extension.logo_url.small,
+    appInstallation(action) {
```

Comment:
It doesn't look like we get any feedback in the event that `appInstallation` fails. My initial tests reveal that the button state transitions from `Install` => `Installing` => `Install`, but I don't get any on screen feedback that an error occurred.

Do we have a pattern defined for relaying errors to the user?